backend/tests/bulkAppointmentsClashes.test.js

```diff
@@ -0,0 +1,376 @@
+              startTime: '2019-09-23T15:30:00',
+            },
+            appointments: [{ bookingId: '333' }, { bookingId: '444' }],
+          })
+
+          expect(res.redirect).toBeCalledWith('/bulk-appointments/appointments-added')
+        })
+      })
+
+      describe('and there are recurring appointments', () => {
+        beforeEach(() => {
+          elite2Api.addBulkAppointments = jest.fn().mockReturnValue('All good')
+          req.session.data = {
+            ...appointmentDetails,
+            recurring: 'yes',
+            times: '5',
+            repeats: 'WEEKLY',
+          }
+        })
+
+        it('should submit the correct data and redirect to the appointments added page', async () => {
+          await controller.post(req, res)
+
+          expect(elite2Api.addBulkAppointments).toBeCalledWith(res.locals, {
+            appointmentDefaults: {
+              appointmentType: 'TEST',
+              comment: 'Activity comment',
+              locationId: 1,
+              startTime: '2019-09-23T15:30:00',
+              endTime: '2019-09-30T16:30:00',
+            },
+            appointments: [{ bookingId: '111' }, { bookingId: '222' }, { bookingId: '333' }, { bookingId: '444' }],
+            repeat: {
+              count: 5,
+              repeatPeriod: 'WEEKLY',
+            },
+          })
+
+          expect(res.redirect).toBeCalledWith('/bulk-appointments/appointments-added')
+        })
+      })
+    })
+
+    describe('and there are individual start and end times', () => {
+      beforeEach(() => {
+        elite2Api.addBulkAppointments = jest.fn().mockReturnValue('All good')
+        req.session.data = {
+          appointmentType: 'TEST',
+          location: 1,
+          date: '27/09/2019',
+          sameTimeAppointments: 'no',
+          comments: 'Activity comment',
+          prisonersNotFound: [],
+          prisonersListed: [
+            {
+              bookingId: '111',
+              offenderNo: 'G1683VN',
+              firstName: 'Elton',
+              lastName: 'Abbatiello',
+              startTimeHours: '18',
+              startTimeMinutes: '45',
+              endTimeHours: '19',
+              endTimeMinutes: '10',
+              startTime: '2019-10-08T18:45:00',
+              endTime: '2019-10-08T19:10:00',
+            },
+            {
+              bookingId: '222',
+              offenderNo: 'G4803UT',
+              firstName: 'Bobby',
+              lastName: 'Abdulkadir',
+              startTimeHours: '18',
+              startTimeMinutes: '50',
+              endTimeHours: '19',
+              endTimeMinutes: '15',
+              startTime: '2019-10-08T18:50:00',
+              endTime: '2019-10-08T19:15:00',
+            },
+          ],
+        }
+      })
+
+      it('submit the data and redirect to the appointments added page', async () => {
+        jest.spyOn(Date, 'now').mockImplementation(() => 1569481200000) // Thursday 2019-09-26T07:00:00.000Z
```

Comment:
Is this needed? Not sure we're doing any new instances of date times/moment() in the controller class? Sorry if I've missed it.

Reply:
Yeah, when we do the supply a default start time when there are individual offender start times.